shell/browser/ui/views/client_frame_view_linux.cc

```diff
@@ -0,0 +1,490 @@
+constexpr float kAdwaitaDarkBorderAlpha = 1 - 0.25;
+constexpr SkColor kAdwaitaLightBorderColor =
+    SkColorSetA(SK_ColorBLACK, 255 * kAdwaitaLightBorderAlpha);
+constexpr SkColor kAdwaitaDarkBorderColor =
+    SkColorSetA(SK_ColorBLACK, 255 * kAdwaitaDarkBorderAlpha);
+
+constexpr int kBorderInset = 1;
+
+}  // namespace
+
+// static
+const char ClientFrameViewLinux::kViewClassName[] = "ClientFrameView";
+
+ClientFrameViewLinux::ClientFrameViewLinux()
+    : nav_button_provider_(
+          views::LinuxUI::instance()->CreateNavButtonProvider()),
+      nav_buttons_{
+          NavButton{views::NavButtonProvider::FrameButtonDisplayType::kClose,
+                    views::FrameButton::kClose, &views::Widget::Close,
+                    IDS_APP_ACCNAME_CLOSE, HTCLOSE},
+          NavButton{views::NavButtonProvider::FrameButtonDisplayType::kMaximize,
+                    views::FrameButton::kMaximize, &views::Widget::Maximize,
+                    IDS_APP_ACCNAME_MAXIMIZE, HTMAXBUTTON},
+          NavButton{views::NavButtonProvider::FrameButtonDisplayType::kRestore,
+                    views::FrameButton::kMaximize, &views::Widget::Restore,
+                    IDS_APP_ACCNAME_RESTORE, HTMAXBUTTON},
+          NavButton{views::NavButtonProvider::FrameButtonDisplayType::kMinimize,
+                    views::FrameButton::kMinimize, &views::Widget::Minimize,
+                    IDS_APP_ACCNAME_MINIMIZE, HTMINBUTTON},
+      },
+      trailing_frame_buttons_{views::FrameButton::kMinimize,
+                              views::FrameButton::kMaximize,
+                              views::FrameButton::kClose} {
+  for (auto& button : nav_buttons_) {
+    button.button = new views::ImageButton();
+    button.button->SetImageVerticalAlignment(views::ImageButton::ALIGN_MIDDLE);
+    button.button->SetAccessibleName(
+        l10n_util::GetStringUTF16(button.accessibility_id));
+  }
+
+  title_ = new views::Label();
+  title_->SetSubpixelRenderingEnabled(false);
+  title_->SetAutoColorReadabilityEnabled(false);
+  title_->SetHorizontalAlignment(gfx::ALIGN_CENTER);
+  title_->SetVerticalAlignment(gfx::ALIGN_MIDDLE);
+  title_->SetTextStyle(views::style::STYLE_TAB_ACTIVE);
+  AddChildView(title_);
+
+  for (NavButton& button : nav_buttons_) {
+    AddChildView(button.button);
+  }
+
+  GetNativeTheme()->AddObserver(this);
+  views::LinuxUI::instance()->AddWindowButtonOrderObserver(this);
+}
+
+ClientFrameViewLinux::~ClientFrameViewLinux() {
+  views::LinuxUI::instance()->RemoveWindowButtonOrderObserver(this);
+  GetNativeTheme()->RemoveObserver(this);
+}
+
+void ClientFrameViewLinux::Init(NativeWindowViews* window,
+                                views::Widget* frame) {
+  FramelessView::Init(window, frame);
+
+  paint_as_active_changed_subscription_ =
+      frame_->RegisterPaintAsActiveChangedCallback(base::BindRepeating(
+          &ClientFrameViewLinux::PaintAsActiveChanged, base::Unretained(this)));
```

Comment:
Are we sure `base::Unretained` is OK here? If so, let's add a short comment here about the reason why it's OK.